### PR TITLE
refactor(repositories): recycled item data extends MutableRepository

### PR DESCRIPTION
**Pull request by Defferrard**

### CODE REVIEW

File: src/services/item/plugins/recycled/repository.ts [hunk 1]
```diff
@@ -1,17 +1,41 @@
...
+  async addOne({ itemPath, creatorId }: CreateRecycledItemDataBody) {
+    return await super.insert({ item: { path: itemPath }, creator: { id: creatorId } });
+  }
+
+  // warning: this call insert in the table
+  // but does not soft delete the item
+  // should we move to core item?
+  async addMany(items: Item[], creator: Member) {
+    const recycled = items.map((item) => this.repository.create({ item, creator }));
+    this.repository.insert(recycled as QueryDeepPartialEntity<RecycledItemData>[]);
```
Comment (ReidyT) [inline src/services/item/plugins/recycled/repository.ts:31]: you have to await here

File: src/services/item/plugins/recycled/repository.ts [hunk 2]
```diff
@@ -1,17 +1,41 @@
...
+    return recycled;
+  }
+
+  async getManyByMember(member: Member): Promise<RecycledItemData[]> {
```
Comment (ReidyT) [inline src/services/item/plugins/recycled/repository.ts:35]: is `getByMember` enough if we choose to specify `getOne...` when returning a single entity?

File: src/services/item/plugins/recycled/repository.ts [hunk 3]
```diff
@@ -25,52 +49,17 @@ export const RecycledItemDataRepository = AppDataSource.getRepository(RecycledIt
...
-    const recycled = items.map((item) => this.create({ item, creator }));
-    this.insert(recycled);
-    return recycled;
-  },
-
-  // warning: this call insert in the table
-  // but does not soft delete the item
-  // should we move to core item?
-  async restoreOne(item: Item) {
+  async deleteManyByItemPath(itemsPath: Item['path'][]) {
```
Comment (ReidyT) [inline src/services/item/plugins/recycled/repository.ts:57]: Same comment here?
```suggestion
  async deleteByItemPath(itemsPath: Item['path'][]) {
```

**Review by ReidyT — APPROVED** [review]
LGTM 🎃 I know that it's not you, but I have a global question, what is the purpose of this repository? I have the feeling that we have to manage two source of truth we that repo + the item repository. Indeed, when we soft remove an item, we also have to add in this repo, and if we restore or delete permanently, we have to use these two repositories right? @spaenleh do you know why this repository is needed?

File: src/services/item/plugins/recycled/repository.ts (hunk 2 above)
Comment (pyphilia) [reply src/services/item/plugins/recycled/repository.ts:35]: I think I like it with the many precision.

**Review by pyphilia — APPROVED** [review]
LGTM!

File: src/repositories/ImmutableRepository.ts [hunk 4]
```diff
@@ -104,7 +104,7 @@ export abstract class ImmutableRepository<T extends BaseEntity> extends Abstract
       throw new IllegalArgumentException(`The given ${name} is undefined!`);
     }
 
-    if (Array.isArray(value) && value.length === 0) {
+    if (Array.isArray(value) && value.length === 0 && value.every((v) => !v)) {
```
Comment (spaenleh) [inline src/repositories/ImmutableRepository.ts:107]: I am not sure to understand how the `value` array could have values inside when the length is zero ...

File: src/services/item/plugins/recycled/repository.ts (hunk 1 above)
Comment (spaenleh) [inline src/services/item/plugins/recycled/repository.ts:31]: Is the casting fixed by an update of the sdk ? I think I removed the `extra.app.setting: unknown` property last week ?

File: src/services/item/plugins/recycled/repository.ts [hunk 5]
```diff
@@ -25,52 +49,17 @@ export const RecycledItemDataRepository = AppDataSource.getRepository(RecycledIt
         { permission: PermissionLevel.Admin, accountId: member.id },
       )
       .getMany();
-
-    return recycledItemEntries;
-  },
+  }
 
   // warning: this call insert in the table
```
Comment (spaenleh) [inline src/services/item/plugins/recycled/repository.ts:54]:
```suggestion
  // warning: this call removes from the table
```

**Review by spaenleh — APPROVED** [review]

**Comment by Defferrard** [comment]
> LGTM 🎃 I know that it's not you, but I have a global question, what is the purpose of this repository? I have the feeling that we have to manage two source of truth we that repo + the item repository. Indeed, when we soft remove an item, we also have to add in this repo, and if we restore or delete permanently, we have to use these two repositories right? @spaenleh do you know why this repository is needed?

I've created an issue for that
https://github.com/graasp/graasp/issues/1345

**Pull request merged**